packages/serverpod/lib/src/server/tasks/task.dart

```diff
@@ -0,0 +1,23 @@
+/// A task that can be executed by the [TaskManager].
+///
+/// Tasks are managed by the [TaskManager] and can be executed at various points,
+/// such as during server shutdown. Tasks can be executed in parallel or
+/// sequentially based on their configuration.
+class Task {
+  /// Creates a new task.
+  ///
+  /// The [id] is used for identification and logging purposes.
+  /// The [callback] is the function that will be executed when the task is run.
+  Task(this.id, this.callback);
+
+  /// The identifier for this task.
+  ///
+  /// Used for identification and logging purposes, especially when errors
+  /// occur.
+  final Object id;
```

Comment:
Given that the `id` has this "uniqueness" property, I wonder if a second `debugLabel` is warranted?

But might be fine without it, in practice I also don't see much removal of tasks, so a more descriptive ID might be just fine :)